Bug: link to media posted on channel = Server Failed to Load

## My Setup
 - __My OS name and version__: OS X or any other...
 - __My current app version__: 2.5.0

- [x] I have tested with the latest application version
- [x] I can simulate the issue easily

## Description

Bug while pasting media link to public channel (for ex. youtube movies) - "Server Failed to Load" - ONLY in Rocket app client. Just can't join again to this channel until the post with link is removed. No errors in server logs. This happens only in public/private channels, priv msg working just fine. Refreshing/clearing cache app doesn't help.

In server dev tools:

Uncaught ReferenceError: Meteor is not defined
    at /Applications/Rocket.Chat+.app/Contents/Resources/app.asar/app/public/preload.js:19
12/Applications/Rocket.Chat+.app/Contents/Resources/app.asar/app/public/preload.js:75 Error getting system idle time: ReferenceError: Meteor is not defined

In app dev tools:

webview.js:75webview: Uncaught TypeError: Cannot read property 'style' of null
webview.js:75webview: Uncaught ReferenceError: Meteor is not defined
13webview.js:75webview: Error getting system idle time: ReferenceError: Meteor is not defined
webview.js:75 webview: Error getting system idle time: ReferenceError: Meteor is not defined

#### Current Behavior
![screen](https://cloud.githubusercontent.com/assets/3985913/23653469/f8d9d6a2-032c-11e7-97e6-8c655985a2ac.jpg)

--- a/src/scripts/webview.js
+++ b/src/scripts/webview.js
@@ -120,6 +120,12 @@ class WebView extends EventEmitter {
             }
         });
 
+        webviewObj.addEventListener('did-get-response-details', (e) => {
+            if (e.resourceType === 'mainFrame' && e.httpResponseCode >= 500) {
+                webviewObj.loadURL('file://' + __dirname + '/loading-error.html');
+            }
+        });
+
         this.webviewParentElement.appendChild(webviewObj);
 
         webviewObj.src = host.lastPath || host.url;

--- a/src/scripts/webview.js
+++ b/src/scripts/webview.js
@@ -114,8 +114,10 @@ class WebView extends EventEmitter {
             this.emit('dom-ready', host.url);
         });
 
-        webviewObj.addEventListener('did-fail-load', () => {
-            webviewObj.loadURL('file://' + __dirname + '/loading-error.html');
+        webviewObj.addEventListener('did-fail-load', (e) => {
+            if (e.isMainFrame) {
+                webviewObj.loadURL('file://' + __dirname + '/loading-error.html');
+            }
         });
 
         this.webviewParentElement.appendChild(webviewObj);

